src/nvim/os/pty_process.c

```diff
@@ -112,13 +107,20 @@ bool pty_process_spawn(Job *job)
 
 void pty_process_close(Job *job)
 {
-  PtyProcess *ptyproc = job->process;
-  uv_signal_stop(&ptyproc->schld);
-  uv_close((uv_handle_t *)&ptyproc->schld, NULL);
+  pty_process_close_master(job);
   job_close_streams(job);
   job_decref(job);
 }
 
+void pty_process_close_master(Job *job)
+{
+  PtyProcess *ptyproc = job->process;
+  if (ptyproc->tty_fd) {
```

Comment:
Is this the correct test to free the master handle? Also `pty_process_spawn()` can return early without initializing `ptyproc->tty_fd`.